src/components/Common/ToggleSwitch.tsx

```diff
@@ -0,0 +1,59 @@
+"use client";
+import { useAppStore } from "@/hooks/useAppStore";
+import { useEffect, useState } from "react";
+
+type ToggleSwitchProps = {
+    label: string;
+};
+
+export function ToggleSwitch({ label }: Readonly<ToggleSwitchProps>) {
+    const { showFavouriteStations, setShowFavouriteStations } = useAppStore();
+
+    const [isMounted, setIsMounted] = useState(false);
+
+    useEffect(() => {
+        setIsMounted(true);
+    }, []);
+
+    if (!isMounted) {
+        return null; // Ensure consistent server-client output
+    }
```

Comment:
I don't get the why here. I believe it has something to do with React/next rendering but I am afraid I don't understand it.

Reply:
As discussed this was added to resolve compilation issues due to using the localstorage in the client side before it was properly initialised on the server side. We're completely removing the toggle in the new approach and as as discussed might be better to use the persist store so I'll explore that instead.